skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import division
+import numpy as np
+from scipy.ndimage.filters import gaussian_filter
+from skimage.util.dtype import dtype_limits
+from skimage import img_as_float
+
+
+def _unsharp_mask_single_channel(image, radius, amount, vrange):
+    """Single channel implementation of the unsharp masking filter."""
+
+    blurred = gaussian_filter(image,
+                              sigma=radius,
+                              mode='reflect')
+
+    result = image + (image - blurred) * amount
+    if vrange is not None:
+        return np.clip(image, vrange[0], vrange[1], out=result)
+    return result
+
+
+def unsharp_mask(image, radius=1.0, amount=1.0, multichannel=False, preserve_range=False):
+    """Unsharp masking filter.
+
+    The sharp details are identified as the difference between the original
+    image and its blurred version. These details are then scaled, and added
+    back to the original image.
+
+    Parameters
+    ----------
+    image : ndarray
```

Comment:
It is better to specify the supported shapes of `image`.
The notation is the following: `[P, ..., ]M[, N][, C]`, i.e. the line should be `image : [P, ..., ]M[, N][, C] ndarray`. Same for `output`.

Reply:
I do not follow the suggested notation.
What do you mean by the middle M?
In my eyes, the algorithm has the same dimension for output as the input was, and the input could be: [D1,D2,....,Dn][,C]

Reply:
The "cypher" from my message above is, basically, meant to explain to the user the range of supported image/ndarray shapes, and the order of different dimensions (e.g. that _channel_ dimension is the last one). The function you developed actually supports not only 2D grayscale or multichannel images, but also 1D grayscale/multichannel arrays and 3D+. This information could come very useful to the users.
Currently (at least, as I perceive it), `M` fuzzily stands for _rows_, `N` for _columns_, `C` for _channels_, and `P, etc` for any extra spatial dimensions.
I'd highly recommend you to take a look at the discussions in https://github.com/scikit-image/scikit-image/issues/2276, https://github.com/scikit-image/scikit-image/pull/3031.